tensorpipe/test/test_environment.h

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#if TENSORPIPE_SUPPORTS_CUDA
+#include <tensorpipe/common/cuda.h>
+#endif // TENSORPIPE_SUPPORTS_CUDA
+
+class TestEnvironment {
+ public:
+  static int numCudaDevices();
```

Comment:
I'm wondering if we should rather make this a member function (a method) rather than a static one. For now a static function makes sense, so that we don't have to pass an instance of the environment around everywhere, but perhaps one day once we add more fields to this we'll want to run a test multiple time with different environments. (In general I guess this relates to my tendency to avoid global state as much as possible).

TL;DR: this is perfectly fine for now, but there's food for thought.